en/02_Developer_Guides/02_Controllers/07_CMS_JSON_APIs.md

```diff
@@ -0,0 +1,154 @@
+
+Use the request param `$ItemID` if you need a record ID into a URL so that you have an endpoint for a specific record. Use `$ItemID` because it's consistent with the request param used in Form Schema requests. For example, to use `$ItemID` in a GET request to view a single record:
+
+```php
+// MySomethingController.php
+namespace App\Controllers\MySomethingController;
+
+use SilverStripe\Admin\LeftAndMain;
+use SilverStripe\Control\HTTPResponse;
+
+class MySomethingController extends LeftAndMain
+{
+    // ...
+
+    private static array $url_handlers = [
+        'GET view/$ItemID' => 'apiView',
+    ];
+
+    public function apiView(): HTTPResponse
+    {
+        $itemID = $request->param('ItemID');
+        // Note: would normally validate that $itemID is a valid integer and that $obj exists
+        $obj = MyDataObject::get()->byID($itemID);
+        $data = ['ID' => $obj->ID, 'Title' => $obj->Title];
+        return $this->jsonSuccess(200, $data);
+    }
+}
+```
+
+Remember to add all public methods that are used as endpoints to `private static array $allowed_actions`.
+
+## Permission checks
+
+Incorporate essential permission checks, such as `canEdit()`, into all relevant endpoints to ensure secure access control.
+
+When returning DataObjects as JSON, remember to invoke `canView()` on each DataObject. In a CMS context where the number of DataObjects is typically limited, the performance impact of these checks should not be a significant concern. If the permission check fails then call `$this->jsonError(403);` to return a 403 status code.
+
+## Return values and error handling
+
+All public endpoint methods must declare a return type of [`HTTPResponse`](api:SilverStripe\Control\HTTPResponse).
+
+All return values must utilise `jsonSuccess()` to create the `HTTPResponse` as this method is used to standardise JSON responses in the CMS.
+
+Do not throw exceptions in the controller, as this leads to a poor content-editor experience. Instead all non-success conditions must call `jsonError()`.
+
+### Using `jsonSuccess()`
+
+When using the optional `$data` parameter in `jsonSuccess()` to return JSON in the response body, do not add any "success metadata" around it, for example `['success' => true, 'data' => $data]`. Instead, solely rely on standard HTTP status codes to clearly indicate the success of the operation.
+
+For scenarios where no JSON data is returned in the response body upon success, use the status code 201 without the `$data` parameter i.e. `return $this->jsonSuccess(201);`. Alternatively, when the response includes JSON data, usually return a 200 status code i.e. `return $this->jsonSuccess(200, $data);`.
+
+### Using `jsonError()`
+
+The `return` keyword is omitted for `jsonError()` because internally it triggers an exception, subsequently caught and converted into an `HTTPResponse` object.
```

Comment:
```suggestion
When calling `jsonError()`, you don't need to use the `return` keyword because that method throws an exception. The exception gets caught in [`handleRequest()`](api:SilverStripe\Control\RequestHandler::handleRequest()) and converted into an `HTTPResponse` object.
```

Reply:
Updated